sympy/physics/units/quantities.py

```diff
@@ -89,7 +89,7 @@ def _eval_is_constant(self):
         return self.scale_factor.is_constant()
 
     def _eval_Abs(self):
-        return self.__class__(self.name, self.dimension,
+        return Quantity(self.name, self.dimension,
```

Comment:
SymPy classes should absolutely support subclassing. @jirikuncar is right here, methods on a class should reference the class name only indirectly, so that subclasses work. Regarding `func` vs. `__class__`, it doesn't matter if the class in question (`Quantity`) keeps them equal.